MODULE_ACCEPTANCE_CRITERIA.MD

```diff
@@ -58,8 +58,13 @@ Please see [Before Development](MODULE_EVALUATION_TEMPLATE#before-development) f
   * _This is not applicable to libraries_
 * Must not depend on a FOLIO library that has not been approved through the TCR process
 * Gracefully handles the absence of third party systems or related configuration. (3, 5, 12)
-* Sonarqube hasn't identified any security issues, major code smells or excessive (>3%) duplication (6); and any disabled or intentionally ignored rules/recommendations are reasonably justified.
+* An automated code scannner hasn't identified any security issues, major code smells or excessive (>3%) duplication (6); and any disabled or intentionally ignored rules/recommendations are reasonably justified.
```

Comment:
How will this automated code scan be run? (currently, Sonar is run by the centralised build process)

How will evidence of the results be supplied to the TC? (currently the results are published to sonar cloud)

Reply:
> If I'm understanding the expanded explanation below, we are stating that any module that uses a language not compatible with Sonar is effectively forced to implement a build within github actions, because there is no other mechanism that can reasonably be used during builds

That would be my understanding as well, and I guess would just be the overhead of introducing a new language that's not Sonar-compatible.  I'm not clear how feasible that is though, given I think we have a shared build process across modules?  So it would have to account for multiple options based on language or something.

Reply:
| we are stating that any module that uses a language not compatible with Sonar is effectively forced to implement a build within github actions, because there is no other mechanism that can reasonably be used during builds
effectively: yes

Reply:
> How will evidence of the results be supplied to the TC?

It suffices to give the "metrics" (=characteristic numbers) as mentioned. Those might be supplied in a YAML file, but does not need to be. See the examples on the page [Examples+of+Code+Duplication...](https://folio-org.atlassian.net/wiki/spaces/TC/pages/386891824/Examples+of+Code+Duplication+Metrics+and+Code+Smells+Severity) (we might need to give more examples)

Reply:
> I guess would just be the overhead of introducing a new language that's not Sonar-compatible.

As far as I recall, the Grails/Groovy modules are not Sonar-compatible. Therefore I changed this sentence: "Sonarqube" -> "An automated code scannner" . In the group discussions I think we agreed that it is in the development teams' responsibility to implement automated test (also for the non Sonar-compatible modules).

Reply:
> I'm not clear how feasible that is though, given I think we have a shared build process across modules? So it would have to account for multiple options based on language or something

That was going to be my next question

I don't know if the back end modules are using github actions

If they are, then I imagine at the very least, the team will need to implement a custom build definition / extensions to use a different static code analysis tool

Reply:
> In the group discussions I think we agreed that it is in the development teams' responsibility to implement automated test

When I read automated test, I think of tests that demonstrate / check the behaviour of the code under test e.g. the module implementation

I don't think of static analysis as a test

@inkuss  In this context, does `automated test` refer to static code analysis (e.g. duplication, security issues etc)?

Reply:
> I don't know if the back end modules are using github actions

I think @julianladisch has more details here.

> I don't think of static analysis as a test, rather it is a separate activity to determine some metrics / characteristics of the code

I agree. Maybe "test" is the wrong word here.
"automated tests" refers solely to static code analysis here. 
I assumed unit tests and integration test are part of static code analysis - maybe wrong assumption ??